### ISSUE TITLE
time to improve the data storage system?

### ISSUE THREAD

**Issue by khlfkk** [bensku/Skript#1168]
csv file is not good design for storage , the files keep growing in size, although i know no one care and will make it lowest  priority , hope there are big change to 2018 for data storage system. Thanks.

**Comment by HYPExMon5ter** [comment]
Yeah, that’s never gonna happen, loading and writing to/from a csv is fast

**Comment by khlfkk** [comment]
true but sad , it will make the csv file size really bigger.

**Comment by Pikachu920** [comment]
nobody really thinks that csv is good choice, its just what njol decided to use. If you look, there is actually a variablesv2 branch

**Comment by khlfkk** [comment]
@Pikachu920 may i have the variablesv2 branch link ? thanks

**Comment by Pikachu920** [comment]
https://github.com/bensku/Skript/tree/feature/variablesv2

**Comment by khlfkk** [comment]
@Pikachu920 thankyou

**Comment by bensku** [comment]
Variables rework was left in very early stage when I last worked on it. Not sure when/if I pick it up again, but definitely not very soon.

**Comment by khlfkk** [comment]
@bensku thanks! although less people concern it , hope make a change to it in 2018

**Comment by khlfkk** [comment]
@bensku 

add "test" to {sql1_.Test::*} #work
add 1 to {sql1_.Test::data::*} #don't work

wait 1 seconds
delete {sql1_.Test::*}

in sqlite database only can delete this {sql1.Test::*} but not {sql1.Test::data::*}  when delete {sql1_.Test::*}
Is it a bug or normal ? thanks

**Comment by bensku** [comment]
Open a new issue about it, I have no idea if that is a bug.

**Comment by khlfkk** [comment]
@bensku thanks for support, but i am tired since i have try many time and it still cause the issue, i will come back to CSV.

**Comment by TheLimeGlass** [comment]
Started working on the variable 2.0 API based on what Bensku started https://github.com/SkriptLang/Skript/tree/feature/variables2
